projects/plugins/jetpack/modules/likes/jetpack-likes-settings.php

```diff
@@ -652,8 +653,17 @@ public function is_comments_enabled() {
 	public function admin_settings_callback() {
 		// We're looking for these, and doing a dance to set some stats and save
 		// them together in array option.
-		$new_state         = ! empty( $_POST['wpl_default'] ) ? sanitize_text_field( wp_unslash( $_POST['wpl_default'] ) ) : 'on'; // phpcs:ignore WordPress.Security.NonceVerification.Missing -- no site changes.
-		$reblogs_new_state = ! empty( $_POST['jetpack_reblogs_enabled'] ) ? sanitize_text_field( wp_unslash( $_POST['jetpack_reblogs_enabled'] ) ) : 'on'; // phpcs:ignore WordPress.Security.NonceVerification.Missing -- no site changes.
+		if ( ! empty( $_POST['wpl_default'] ) && isset( $_POST['_wpnonce'] ) && wp_verify_nonce( $_POST['_wpnonce'], 'sharing-options' ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput -- WordPress core doesn't unslash or verify nonces either.
+			$new_state = sanitize_text_field( wp_unslash( $_POST['wpl_default'] ) );
+		} else {
+			$new_state = 'on';
```

Comment:
The logic here seems wrong. If the nonce check fails, we change the setting to "on"?

In practice it looks like an earlier nonce check will have already prevented this from being called so that won't actually happen. But if we're going to have a nonce check at all instead of relying on that earlier one, we should do the check separately and either `return` or or `wp_die()` depending on which behavior makes more sense.